src/components/about-us/section-05/index.js

```diff
@@ -0,0 +1,465 @@
+   * @param {number} currentYear
+   */
+  _getYear = (currentYearIndex) => {
+    this.setState({
+      currentYear: yearList[currentYearIndex]
+    })
+  }
+  /**
+   * The function is used by list.js for computing the height of children in timeline after they are mounted
+   * @param {Array} contentHeightArray
+   */
+  _getYearContentHeight = (contentHeightArray) => {
+    this.setState({ yearContentHeight: contentHeightArray })
+  }
+  componentDidMount() {
+    const timelineScrollingHeight = ReactDOM.findDOMNode(this.scrollingContent).getBoundingClientRect().height
+    this.setState({ timelineScrollingHeight: timelineScrollingHeight })
+  }
+  render() {
+    return (
+      <Container>
+        <SectionWrapper>
+          <Title><span>大事紀</span></Title>
+          <Circle color={`${colors.black}`}>
+            <Circle color={`${colors.gray.gray96}`}/>
+            <Rect color={`${colors.secondaryColor}`}/>
+          </Circle>
+          <Content>
+            <AccordionTimeline>
+              <List
+                unfoldArray={this.state.unfoldArray}
+                sortedData={sortedData}
+                sortedDataGroupByYear={sortedDataGroupByYear}
+                foldAndUnfold={this._foldAndUnfold}
+                getYearContentHeight={this._getYearContentHeight}  
+              />
+            </AccordionTimeline>
+            <YearRange>
+              <p><span>{yearList[yearList.length - 1]}</span></p>
+              <p>
+                <span>{yearList[0]}</span>
+                <img src={arrow} />
+              </p>
+            </YearRange>
+            <RunningTimeline>
+              <Timeline
+                ref={scrollingContent => this.scrollingContent = scrollingContent}
+                childrenHeight={this.state.timelineScrollingHeight * timelineScrollingPortion} 
+                autoScrolling={this.state.timelineScrolling}
+                startAutoScroll={this._startTimelineAutoScrolling}
+                stopAutoScroll={this._stopTimelineAutoScrolling}
+                yearContentHeight={this.state.yearContentHeight}
+                getYear={this._getYear}>
+                <List
+                  unfoldArray={this.state.unfoldArray}
+                  sortedData={sortedData}
+                  sortedDataGroupByYear={sortedDataGroupByYear}
+                  foldAndUnfold={this._foldAndUnfold}
+                  getYearContentHeight={this._getYearContentHeight}  
+                />
+              </Timeline>
+            </RunningTimeline>
+            <YearTag>
+              <p>{this.state.currentYear}</p>
+            </YearTag>
+          </Content>
+        </SectionWrapper>
+        <BorderBottom 
+          fixed={this.state.isBorderBottomfixed}
+          zIndex={this._getBorderZIndex()}
+        />
+        <Waypoint
+          onPositionChange={({ previousPosition, currentPosition }) => this._onPositionChange(previousPosition, currentPosition)}
```

Comment:
use `onEnter` and `onLeave` ?
`onPositionChange` has too many conditions including `Waypoint.inside`, `Waypoint.below`, `Waypoint.above` and `Waypoint.invisible`. `onPositionChange` is needed in other complicated situations like you want to know the scrolling direction.

Reply:
BTW, you should stop auto-scrolling when section-05 is not in the viewport or the device is `mobile`.

Reply:
In section5, It is really a good idea to change the `onPositionChange` to `onEnter` and `onLeave` callback function, but I'll keep `onPositionChange` in section1 since I need to know the scrolling direction to decide whether the position of border-top is `fixed` or `absolute`